src/coreclr/vm/jitinterface.cpp

```diff
@@ -2855,6 +2889,53 @@ void CEEInfo::ScanTokenForDynamicScope(CORINFO_RESOLVED_TOKEN * pResolvedToken,
     ScanToken(pModule, pResolvedToken, th, pMD);
 }
 
+OBJECTHANDLE CEEInfo::getJitHandleForObject(OBJECTREF obj)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_COOPERATIVE;
+    }
+    CONTRACTL_END;
+
+    if (m_pJitHandles == nullptr)
+    {
+        m_pJitHandles = new SArray<OBJECTHANDLE>();
+    }
+
+    OBJECTHANDLE handle;
+    GCPROTECT_BEGIN(obj);
+    handle = AppDomain::GetCurrentDomain()->CreateHandle(obj);
+
+    // We know that handle is aligned so we use the lowest bit as a marker
+    // "this is a handle, not a frozen object".
+    handle = (OBJECTHANDLE)((size_t)handle | 1);
+    m_pJitHandles->Append(handle);
+    GCPROTECT_END();
+    return handle;
+}
+
+Object* CEEInfo::getObjectFromJitHandle(OBJECTHANDLE handle)
```

Comment:
It would be better for this to return `OBJECTREF`. `OBJECTREF`s have additional sanity checks in debug builds. They are the preferred type to pass `Object*` around in the VM.